integration/harness.go

```diff
@@ -69,23 +68,23 @@ func NewHarness(cfg *ServerConfig) (*Harness, error) {
 // The created process is killed, and the temporary
 // directories are removed.
 func (h *Harness) TearDown(cleanup bool) error {
-	if err := h.server.shutdown(cleanup); err != nil {
-		return err
+	if err := h.conn.Close(); err != nil {
+		return fmt.Errorf("failed to close connection: %w", err)
```

Comment:
Is it alright to return early, even in case `conn` failed to close? IMO, it should optimistically continue shutting down other things. Especially, there are still some things to `cleanup`.

Reply:
It's arguably not an error if the cleanup still works, so instead of returning the error I just log it.

Reply:
I don't understand.

The code just returns if `h.conn.Close()` failed. My point is that it should store the error and try to proceed with shutdown. Finally return any error it saw during Teardown.

Reply:
No it doesn't, please check again I already replaced the `return` with a `log`